From 27744be887481f9659e0dc4491400dd78338661a Mon Sep 17 00:00:00 2001
From: Alex R <alex@ironrobin.net>
Date: Sun, 5 Oct 2025 19:03:38 +0000
Subject: [PATCH] InputDispatchFilter:Wheel: take Shell's rotation into account

---
 plugins/Cursor/InputDispatcherFilter.cpp | 15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/plugins/Cursor/InputDispatcherFilter.cpp b/plugins/Cursor/InputDispatcherFilter.cpp
index 7253a3bce..0e301c6d8 100644
--- a/plugins/Cursor/InputDispatcherFilter.cpp
+++ b/plugins/Cursor/InputDispatcherFilter.cpp
@@ -143,14 +143,25 @@ bool InputDispatcherFilter::eventFilter(QObject *o, QEvent *e)
             auto pointer = currentPointer();
             if (!pointer || !pointer->window()) return true;
 
+            if (!pointer->parentItem()) {
+                qDebug() << "Huh? a MousePointer without a parent?";
+                return true;
+            }
+
             QWheelEvent* we = static_cast<QWheelEvent*>(e);
 
             // Local position gives relative change of mouse pointer.
             QPointF movement = we->position();
 
+            // Because the movement is relative to the visual display, run both old and new
+            // pointer position through mapToGlobal() to take Shell's rotation into account,
+            // then recalculate the movement as a difference between them.
+            QPointF oldPos = pointer->parentItem()->mapToGlobal(pointer->position());
+            QPointF newPos = pointer->parentItem()->mapToGlobal(pointer->position() + movement);
+            movement = newPos - oldPos;
+
             // Adjust the position
-            QPointF oldPos = pointer->window()->geometry().topLeft() + pointer->position();
-            QPointF newPos = adjustedPositionForMovement(oldPos, movement);
+            newPos = adjustedPositionForMovement(oldPos, movement);
 
             // Send the event
             QWheelEvent eCopy(we->position(), newPos, we->pixelDelta(), we->angleDelta(), we->buttons(), we->modifiers(), we->phase(), we->inverted());
