From 7c9874f515bb224c6de7410c44efe4a2980aee55 Mon Sep 17 00:00:00 2001
From: Alex R <alex@ironrobin.net>
Date: Tue, 7 Oct 2025 23:34:41 +0000
Subject: [PATCH] PATCH: add touchpad tap to click

---
 src/platforms/mirserver/qmirserver_p.cpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/platforms/mirserver/qmirserver_p.cpp b/src/platforms/mirserver/qmirserver_p.cpp
index 3aef77d8..759e22b0 100644
--- a/src/platforms/mirserver/qmirserver_p.cpp
+++ b/src/platforms/mirserver/qmirserver_p.cpp
@@ -30,6 +30,7 @@
 #include "promptsessionlistener.h"
 #include "qtwindowmanager.h"
 #include "wayland/kwinserverdecoration.h"
+#include <miral/input_configuration.h>
 
 #include <mir/version.h>
 #if MIR_SERVER_VERSION >= MIR_VERSION_NUMBER(2, 16, 0)
@@ -194,6 +195,12 @@ void QMirServerPrivate::run(const std::function<void()> &startCallback)
     waylandExtensions.add_extension(qtmir::qtWindowmanagerExtension());
     waylandExtensions.add_extension(qtmir::qtKwinServerDecorationManagerExtension(m_windowModelNotifier));
 
+    miral::InputConfiguration input_configuration;
+
+    auto touchpad = input_configuration.touchpad();
+    touchpad.tap_to_click(true);
+    input_configuration.touchpad(touchpad);
+
     runner.set_exception_handler([]
     {
         try {
@@ -213,6 +220,7 @@ void QMirServerPrivate::run(const std::function<void()> &startCallback)
         {
             setenv("DISPLAY", x11_display.value().c_str(), 1);
         }
+        input_configuration.touchpad(touchpad);
     });
 
     runner.add_start_callback(startCallback);
@@ -278,6 +286,7 @@ void QMirServerPrivate::run(const std::function<void()> &startCallback)
             ),
             setTerminator,
             config_keymap,
+            std::ref(input_configuration),
             miroil::PersistDisplayConfig{displayStorageBuilder(),
                                         m_displayConfigurationPolicy},
         });
